core/frontend/helpers/content.js

```diff
@@ -18,9 +18,26 @@ const createFrame = hbs.handlebars.createFrame;
 function restrictedCta(options) {
     options = options || {};
     options.data = options.data || {};
+
     _.merge(this, {
         accentColor: (options.data.site && options.data.site.accent_color)
     });
+    if (this.visibility.includes('product:')) {
+        const productSlugs = this.visibility.split(',').filter(qry => qry.startsWith('product:')).map(qry => qry.replace('product:', ''));
```

Comment:
If visibility _is_ going to be used to store NQL strings, we should be using our language parser to parse them

Reply:
@ErisDS Ah yea good point! This was just a first version hacked out for demo purposes, but yeah didn't strike me to use language parser as the first thing to use here :)